src/control/security/auth/auth_sys.go

```diff
@@ -46,43 +46,63 @@ type UserExt interface {
 	LookupGroupID(gid uint32) (*user.Group, error)
 }
 
-// HashFromToken will return a SHA512 hash of the token data
-func HashFromToken(token *auth.Token) ([]byte, error) {
-	// Generate our hash (not signed yet just a hash)
-	hash := sha512.New()
-
+//VerifierFromToken will return a SHA512 hash of the token data. If a signing key
+//is passed in it will additionally sign the hash of the token.
+func VerifierFromToken(signingKey crypto.PublicKey, token *Token) ([]byte, error) {
 	tokenBytes, err := proto.Marshal(token)
 	if err != nil {
 		return nil, errors.Wrap(err, "Unable to marshal AuthToken")
 	}
 
-	hash.Write(tokenBytes)
-	hashBytes := hash.Sum(nil)
-	return hashBytes, nil
+	signer := security.DefaultTokenSigner()
```

Comment:
Is there a possibility that you might eventually want these functions to use a different token signer? From an API design standpoint, injecting the signer as part of the function params would give you flexibility, but might be overkill.

Reply:
The signer just takes a byte array so its pretty generic to begin with. Its up to the token to convert itself into the bytes to sign.